src/core/sig_state.ml

```diff
@@ -109,32 +106,32 @@ let add_symbol : sig_state -> sig_symbol -> sig_state * sym =
   ({ss with in_scope; pp_hints}, s)
 
 (** [add_unop ss n x] generates a new signature state from [ss] by adding a
-   unary operator [x] with name [n]. *)
-let add_unop : sig_state -> string -> (sym * unop) -> sig_state =
-  fun ss name ((sym, unop) as x) ->
-  Sign.add_unop ss.signature name x;
-  let unops = StrMap.add name sym ss.unops in
-  let pp_hints = remove_pp_hint ss.unops name ss.pp_hints in
+    unary operator [x] with name [n]. This name is added to the scope. *)
+let add_unop : sig_state -> strloc -> (sym * unop) -> sig_state =
+  fun ss name (sym, unop) ->
+  Sign.add_unop ss.signature sym unop;
+  let in_scope = StrMap.add name.elt (sym, name.pos) ss.in_scope in
+  let pp_hints = remove_pp_hint sym ss.pp_hints in
   let pp_hints = SymMap.add sym (Prefix unop) pp_hints in
-  {ss with unops; pp_hints}
+  {ss with in_scope; pp_hints}
 
 (** [add_binop ss n x] generates a new signature state from [ss] by adding a
-   binary operator [x] with name [n]. *)
-let add_binop : sig_state -> string -> (sym * binop) -> sig_state =
-  fun ss name ((sym, binop) as x) ->
-  Sign.add_binop ss.signature name x;
-  let binops = StrMap.add name sym ss.binops in
-  let pp_hints = remove_pp_hint ss.binops name ss.pp_hints in
+    binary operator [x] with name [n]. This name is added to scope. *)
+let add_binop : sig_state -> strloc -> (sym * binop) -> sig_state =
+  fun ss name (sym, binop) ->
+  Sign.add_binop ss.signature sym binop;
+  let in_scope = StrMap.add name.elt (sym, name.pos) ss.in_scope in
+  let pp_hints = remove_pp_hint sym ss.pp_hints in
   let pp_hints = SymMap.add sym (Infix binop) pp_hints in
```

Comment:
remove/add can be simplified to `SymMap.update sym (function _ -> Some (Infix binop)) ss.pp_hints`.